### PR TITLE
AC-766 Updated all the dependecies to latest versions

### PR DESCRIPTION
**Pull request by rishabh-997**
## Description of what I changed
1. Updated dependency
2. Created a new file to store versions which is standard way. 
3. Few dependencies were repeated so removed them
4. Arranged the dependencies to represent a certain group

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-766

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java [hunk 1]
```diff
@@ -260,7 +260,7 @@ private RecyclerView addRecycleView(List<Patient> patientsList, Patient newPatie
         recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
         recyclerView.setAdapter(new SimilarPatientsRecyclerViewAdapter((getActivity()), patientsList, newPatient));
         mFieldsLayout.addView(field);
-        recyclerView.setHasFixedSize(true);
+        recyclerView.setHasFixedSize(false);
         return recyclerView;
```
Comment (LuGO0) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java:264]: @rishabh-997 just out of curiosity why this change does it fix some ripples caused due to change in dependencies or just normal fix ??

Comment (rishabh-997) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java:264]: actually I updated the recycler view dependency, maybe they have added those lint warning where you cant have fixed size in the scroll direction.

File: openmrs-client/build.gradle [hunk 2]
```diff
@@ -117,15 +118,67 @@ repositories {
...
-    def room_version = "2.1.0-alpha04"
-    def appcompat_version = "1.2.0-alpha01"
-    def material_version = "1.2.0-alpha03"
-    def worker_manager_version = "2.3.4"
-
-    implementation "androidx.work:work-runtime:$worker_manager_version"
-    implementation 'com.google.code.gson:gson:2.8.5'
-    implementation "androidx.appcompat:appcompat:$appcompat_version"
+
+    androidTestImplementation files('libs/the-missing-android-xml-junit-test-runner-release-1.3_2.jar')
```
Comment (f4ww4z) [inline openmrs-client/build.gradle:122]: Looks like this lib is [from here](https://github.com/Polidea/the-missing-android-xml-junit-test-runner) and is very outdated, we should not use it. @rishabh-997 did you try running tests with this removed?

File: openmrs-client/build.gradle [hunk 3]
```diff
@@ -117,15 +118,67 @@ repositories {
...
+    implementation "com.squareup.retrofit2:retrofit:${versions.retrofit}"
+    implementation "com.squareup.retrofit2:converter-gson:${versions.retrofitConverter}"
+    implementation "com.google.code.gson:gson:${versions.gson}"
+
+    // Rx Java
+    implementation "io.reactivex:rxjava:${versions.rxjava}"
+    implementation "io.reactivex:rxandroid:${versions.rxandroid}"
+    implementation "com.jakewharton.rxbinding:rxbinding:${versions.rxbinding}"
+
+    // Room components
```
Comment (f4ww4z) [inline openmrs-client/build.gradle:142]: Rename this to Database components.

File: openmrs-client/build.gradle [hunk 4]
```diff
@@ -117,15 +118,67 @@ repositories {
...
+    implementation "com.google.guava:guava:${versions.guava}"
+
+    // Third Party Performance Libraries
+    implementation "com.github.hotchemi:permissionsdispatcher:${versions.permissionsdispatcher}"
+    annotationProcessor "com.github.hotchemi:permissionsdispatcher-processor:${versions.permissionsdispatcher}"
+    implementation "androidx.multidex:multidex:${versions.multidex}"
+    implementation "joda-time:joda-time:${versions.jodaTime}"
+    implementation "net.sf.kxml:kxml2:${versions.kxml2}"
+    implementation "commons-codec:commons-codec:${versions.codec}"
+    implementation "net.zetetic:android-database-sqlcipher:${versions.sqlcipher}"
```
Comment (f4ww4z) [inline openmrs-client/build.gradle:173]: Move this under Database components.

File: openmrs-client/versions.gradle [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+versions.MPAndroidChart = 'v3.0.2'
+versions.showcaseview = '5.4.3'
+versions.jbcrypt = '0.4'
+versions.countryCodePicker = '2.2.7'
+versions.AppIntro = '5.1.0'
+versions.swipereveallayout = '1.4.1'
+
+versions.codec = '1.10'
+versions.sqlcipher = '4.1.3@aar'
+versions.jdeferred = '1.2.4'
```
Comment (f4ww4z) [inline openmrs-client/versions.gradle:41]: Should [be 1.2.6](https://repo1.maven.org/maven2/org/jdeferred/jdeferred-android-aar/1.2.6/)

File: openmrs-client/versions.gradle [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+versions.roomRuntime = '2.3.0-alpha01'
+versions.roomPersistence = '1.1.1'
+versions.roomCompiler = '2.1.0-alpha04'
+
+versions.chuckLibrary = '1.1.0'
+
+versions.workManager = '2.3.4'
+
+versions.seekbar = '1.0.1'
+versions.MPAndroidChart = 'v3.0.2'
```
Comment (f4ww4z) [inline openmrs-client/versions.gradle:32]: Should be [v3.1.0](https://github.com/PhilJay/MPAndroidChart#gradle-setup)

File: openmrs-client/versions.gradle [hunk 7]
```diff
@@ -0,0 +1,51 @@
+ext.versions = [:]
+
+versions.compileSdk = 29
+versions.minSdk = 16
+versions.targetSdk = 29
+versions.appVersion = "2.8.3"
+
+//Libraries
```
Comment (f4ww4z) [inline openmrs-client/versions.gradle:8]:
```suggestion
// Libraries
```
Just to follow openmrs conventions.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/introduction/IntroActivity.kt [hunk 8]
```diff
@@ -30,28 +30,65 @@ class IntroActivity : AppIntro2() {
...
-        addSlide(AppIntroFragment.newInstance(getString(R.string.intro_register), getString(R.string.intro_register_desc), R.drawable.ico_registry, Color.parseColor("#F26522")))
-        addSlide(AppIntroFragment.newInstance(getString(R.string.intro_find), getString(R.string.intro_find_desc), R.drawable.ico_search, Color.parseColor("#009384")))
-        addSlide(AppIntroFragment.newInstance(getString(R.string.intro_monitor), getString(R.string.intro_monitor_desc), R.drawable.ico_visits, Color.parseColor("#F0A815")))
-        addSlide(AppIntroFragment.newInstance(getString(R.string.intro_manage), getString(R.string.intro_manage_desc), R.drawable.ic_provider_big, Color.parseColor("#F26522")))
-        addSlide(AppIntroFragment.newInstance(getString(R.string.intro_location), getString(R.string.intro_location_desc), R.drawable.ic_location_big, Color.parseColor("#009384")))
-        addSlide(AppIntroFragment.newInstance(getString(R.string.intro_settings), getString(R.string.intro_settings_desc), R.drawable.ic_settings_big, Color.parseColor("#F0A815")))
+        addSlide(AppIntroFragment.newInstance(
```
Comment (rishabh-997) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/introduction/IntroActivity.kt:33]: with the migration of AppIntro from v5 to v6, the entire codebase was converted to kotlin and had to rewrite almost entire fragment.

Comment (f4ww4z) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/introduction/IntroActivity.kt:33]: Do they just require named arguments now? Or was there more to change.
